cpp/src/arrow/flight/sql/odbc/odbcabstraction/calendar_utils.cc

```diff
@@ -39,13 +41,30 @@ int64_t GetTodayTimeFromEpoch() {
 #endif
 }
 
-void GetTimeForSecondsSinceEpoch(tm& date, int64_t value) {
-#if defined(_WIN32)
-  gmtime_s(&date, &value);
-#else
-  time_t time_value = static_cast<time_t>(value);
-  gmtime_r(&time_value, &date);
-#endif
+void GetTimeForSecondsSinceEpoch(std::tm& out_tm, int64_t seconds_since_epoch) {
+  try {
+    std::chrono::time_point<std::chrono::system_clock, std::chrono::seconds> timepoint{
+        std::chrono::seconds{seconds_since_epoch}};
+
+    std::chrono::year_month_day ymd = std::chrono::floor<std::chrono::days>(timepoint);
+
+    std::chrono::hh_mm_ss<std::chrono::seconds> timeofday{
+        timepoint - std::chrono::floor<std::chrono::days>(timepoint)};
+
+    std::memset(&out_tm, 0, sizeof(std::tm));
+
+    out_tm.tm_year = static_cast<int>(ymd.year()) - 1900;
+    out_tm.tm_mon = static_cast<unsigned>(ymd.month()) - 1;
+    out_tm.tm_mday = static_cast<unsigned>(ymd.day());
```

Comment:
Ditto for `tm_mday`

Reply:
Same here https://en.cppreference.com/w/cpp/chrono/day/operator_unsigned.html